lib/matplotlib/collections.py

```diff
@@ -1101,14 +1105,21 @@ def set_verts(self, verts, closed=True):
 
         # Fast path for arrays
         if isinstance(verts, np.ndarray):
-            verts_pad = np.concatenate((verts, verts[:, :1]), axis=1)
+            verts_pad = (np.concatenate((verts, verts[:, :1]), axis=1)
+                           .astype(mpath.Path.verts_type))
             # Creating the codes once is much faster than having Path do it
             # separately each time by passing closed=True.
             codes = np.empty(verts_pad.shape[1], dtype=mpath.Path.code_type)
             codes[:] = mpath.Path.LINETO
             codes[0] = mpath.Path.MOVETO
             codes[-1] = mpath.Path.CLOSEPOLY
-            self._paths = [mpath.Path(xy, codes) for xy in verts_pad]
+            example_path = mpath.Path(verts_pad[0], codes)
+            # Looking up the function once before the iteration gives a nice speedup
+            _make_path = mpath.Path._fast_from_codes_and_verts
+            self._paths = [_make_path(xy, codes,
```

Comment:
Turns out that the Python function call overheads become quite significant here, and so inlining the body of `_fast_from_codes_and_verts` into this function actually results in an _end-to-end 10% improvement_ for my example plot! On the other hand, it's a method that touches all the Path internals, so I would feel a bit weird inlining it here...